### PR TITLE
For next

### PR DESCRIPTION
**Pull request by PeterMacleodThompson**
Update of ozmaps/mapmanager.c - 4 commits:
ozmaps: mapmanager.c: add 4 function prototypes to fix compiler warnings
ozmaps: Remove DEBUG define from mapmanager.c
ozmaps: mapmanager.c end getmap() function properly to remove compiler warnings
ozmaps: mapmanager.c make comments readable where clang-format made unreadable

This fixes mapmanager.c except for 1 compiler error which will be fixed in a later commit.

UPDATE -  Feb 8: 4pm
Another 9 commits were added here to:
 - finish removing DEBUG define
 - removes all compiler errors!!
 - add/change some comments
 - cleanup long descriptive comments which were muddled by clang-format

TODO 
3 new branches (I hope) to keep separate from the above commits:
 - dieface  with 2 commits to prepare for integration of dieface/viewsensor.c into hikingGPS  
       The daemons will remain separate of course. 
 - BBB with 2-3 commits to distinguish beaglebone black compiles via -D BBB
 - SDL2init with 3 commits to 
      replace a deprecated SDL2 command, 
      print SDL2 version,
      move find_datadir() from displaymap.c to SDL2init.c

### CODE REVIEW

File: src/ozmaps/mapmanager.c [hunk 1]
```diff
@@ -60,16 +60,6 @@ int keysubtract(int, int);
...
-//  #define DEBUG
-#ifdef DEBUG
-
-#include "SDL2/SDL2_gfxPrimitives.h"
-#include <SDL2/SDL.h>
-#include <SDL2/SDL_image.h>
-#define SCREEN_WIDTH 500
-#define SCREEN_HEIGHT 500
-
-#endif
```
Comment (eballetbo) [inline src/ozmaps/mapmanager.c]: note: If you follow the clang-format you should also remove one line, so there is only one line between the end of inthemap function and the comment.

File: src/ozmaps/mapmanager.c [hunk 2]
```diff
@@ -248,7 +248,8 @@ SDL_Surface *getmap(int mapflags,      /* instructions on which map to get */
       camera->y = 0;
       return (mapNow);
     }
-  }
+  } else 
+     return (mapNow);  /* should never reach here */
 }
```
Comment (eballetbo) [inline src/ozmaps/mapmanager.c:253]: You didn't run `clang-format -i` after do the modifications.

Comment (eballetbo) [reply src/ozmaps/mapmanager.c:253]: Yeah, you should really fix the format in the commit that introduced the problem, not create a new commit to fix the format.

File: src/ozmaps/mapmisc.c [hunk 3]
```diff
@@ -109,10 +107,6 @@ int gpstogpskey(uint64_t gps) {
     latmm++;
   gpskey = longdd * 1000000 + longmm * 10000 + latdd * 100 + latmm;
 
-#ifdef DEBUG
-  printf("gps = %lu\n", gps);
-  printf("gpskey(dddmmddmm) = %d\n", gpskey);
-#endif
 
   return (gpskey);
 }
```
Comment (eballetbo) [inline src/ozmaps/mapmisc.c:111]: lgtm

File: src/ozdisplay/SDL2misc.c [hunk 4]
```diff
@@ -45,23 +43,7 @@ process...
...
-
-  if (map == NULL)
-    printf("Unable to load image! SDL_image Error: %s\n", SDL_GetError());
-
-  return (map);
-}
-#endif
-
+/*** load spritesheet from file ***/
 /* FIXME memory leak!!! sprite never gets SDL_FreeSurface( sprite ); */
```
Comment (eballetbo) [inline src/ozdisplay/SDL2misc.c:46]: That change is unrelated to the commit

File: src/ozdisplay/SDL2misc.c [hunk 5]
```diff
@@ -77,7 +59,9 @@ SDL_Surface *getsprite(char *filename) {
 
 /*****************   initsprite ***************************
  * requires square sprites layed out length-wise in spritesheet
- * returns number of sprites. Completes array of SDL_Rects for extraction
+ * can be any size (pixels) so long as it is square
+ * number of animations within sprite computed via sprite width/height
+ * returns number of animations. Completes array of SDL_Rects for extraction
  */
```
Comment (eballetbo) [inline src/ozdisplay/SDL2misc.c]: That change is unrelated to the commit

File: src/ozdisplay/displaymap.c [hunk 6]
```diff
@@ -173,8 +171,6 @@ main() {
   Uint8 alphabutton;          // alpha fading for buttons
   char filename[PATH_MAX];
 
-  Uint64 gps;
-  int mymapindex; /* %100 = mapsetindex, /100 = mapindex within mapset*/
 
   /******************** Step 1 game initialize **************************/
```
Comment (eballetbo) [inline src/ozdisplay/displaymap.c:176]: lgtm

File: src/ozdisplay/displaymap.c [hunk 7]
```diff
@@ -112,7 +112,7 @@ static char *find_datadir() {
   return NULL;
 }
 
-main() {
+int main() {
   SDL_Surface *mymap; /*mymap= active map on screen */
   SDL_Surface *spritesheet;
   SDL_Surface *buttonsheet;
```
Comment (eballetbo) [inline src/ozdisplay/displaymap.c:119]: lgtm

File: src/ozdisplay/displaymap.c [hunk 8]
```diff
@@ -74,6 +74,7 @@ int scalepoint(SDL_Rect *, SDL_Point *, SDL_Point *);
 PMTmapset *initmaps(void);
 SDL_Surface *getmap(int, PMTmapset *, SDL_Rect *);
 SDL_Point *placegps(SDL_Surface *mymap);
+void closemaps(PMTmapset *);
 
 char *datapath = NULL;
```
Comment (eballetbo) [inline src/ozdisplay/displaymap.c:80]: lgtm

**Review by eballetbo — CHANGES_REQUESTED** [review]
Hi Peter,

You're doing really well, I did some trivial comments, that again, because I'm strict for now, I'd like you to address, but basically is that you forgot to run clang-format -i before committing the changes.

Can we leave the documentation modifications that you did out from this pull request for now? I'd like to propose a documentation format instead.

So try for now only push the fixes for the warnings.

**Comment by PeterMacleodThompson** [comment]
> Hi Peter,
> 
> You're doing really well, I did some trivial comments, that again, because I'm strict for now, I'd like you to address, but basically is that you forgot to run clang-format -i before committing the changes.
> 
> Can we leave the documentation modifications that you did out from this pull request for now? I'd like to propose a documentation format instead.
> 
> So try for now only push the fixes for the warnings.

Hi Enric,
Thanks for the vote of confidence :)    Still much to learn!!!

Sorry for forgetting the clang-format and putting it in a separate commit later.  I was unsure of how to change history......
 
I am happy remove the documentation modifications.   Not sure how I should change pushed history.   Should I "git revert" the commits (which will add new commits) and then push the new commits that reverse them?
git revert bbc2287  (ozmaps/mapmanager.c)
git revert a62b572 (ozmaps/mapmisc.c)
git revert b137d18 (include/maps.h)

Thanks for your help!!!!
Peter

File: src/ozdisplay/SDL2misc.c (hunk 5 above)
Comment (PeterMacleodThompson) [reply src/ozdisplay/SDL2misc.c]: Yes - it maybe should have been a separate commit.    My lame excuse is that while inside the code, I couldn't remember how sprites worked, and was disgusted with my poor description so after significant work to re-figure sprites out, I changed the comments immediately.   The change was noted in the details of the  commit, but not the header....

**Comment by PeterMacleodThompson** [comment]
Hi Enric,
The documentation modifications have been removed via 7 commits.
This involved  5 reverts, and 2 adds.  
3 reverts removed pure documentation commits, b137d18, ac2b572, bbc2887
1 revert was a clang-format required to resolve a git conflict: 485d33f conflicted with bbc2887
1 revert was a split removal: part was kept and part was deleted, ee777de
1 add was the clang format replacing commit 485d33f 
1 add was a   ee777de repeat but omitting documentation portion 

Peter

**Comment by eballetbo** [comment]
Merged into mainline. Well done!

**Comment by PeterMacleodThompson** [comment]
Excellent - thanks
Now onto bigger and better things!! :)

**Comment by PeterMacleodThompson** [comment]
Hi Enric,
Should we delete the old for-next branch and restart a new for-next branch pointing to current HEAD?    Or do we continue to extend the old existing for-next forever?   

Whatever you think is best is fine with me.    On my local,  I was planning to delete all "accepted" for-next branches so my new for-next is clean. 

Peter
